packages/api/src/hooks/useAuthorize.ts

```diff
@@ -1,14 +1,14 @@
 import { useEffect, useMemo } from 'react';
 import { getActiveAuthTokenIDFromLocalStorage } from '@deriv/utils';
-import useFetch from '../useFetch';
+import useQuery from '../useQuery';
 
 /** A custom hook that authorize the user with the given token. If no token is given,
  * it will use the current token from localStorage.
  */
 const useAuthorize = (token?: string) => {
     const current_token = getActiveAuthTokenIDFromLocalStorage();
 
-    const { data, ...rest } = useFetch('authorize', {
+    const { data, ...rest } = useQuery('authorize', {
         payload: { authorize: token || current_token || '' },
         options: { enabled: Boolean(current_token) },
```

Comment:
@farzin-deriv just want to ask you about this check. I suppose we have to check the final token, not just `current_token`. We can face the case when token was passed, for example, '123' and `current_token` is empty in `localStorage`, in this case we never do API call because it will be `false` in the enabled property.

<img width="680" alt="image" src="https://github.com/binary-com/deriv-app/assets/120570511/540d7bb3-85c3-4a22-8dd2-7047ffccbeda">

Reply:
@sergei-deriv `useAuthorize` has been changed in #10008.